### ISSUE TITLE
[Question] 关于有条件和无条件的数据疑惑

### ISSUE THREAD

**Issue by jxz7110** [PKU-Alignment/align-anything#192]
### Required prerequisites

- [x] I have read the documentation <https://align-anything.readthedocs.io>.
- [x] I have searched the [Issue Tracker](https://github.com/PKU-Alignment/align-anything/issues) and [Discussions](https://github.com/PKU-Alignment/align-anything/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/PKU-Alignment/align-anything/discussions/new).

### Questions

代码似乎有错误，如果是我的问题，希望理解
![Image](https://github.com/user-attachments/assets/995b7803-c44b-4ea9-8299-8d64e6fa2981)

![Image](https://github.com/user-attachments/assets/f896af43-3ef5-4ad7-99aa-2290f2c7becd)在训练过程中发现将与条件数据放在前一半  无条件数据放在后一半  但是后续logits是奇偶放置的

**Comment by Gaiejj** [comment]
非常抱歉带来困惑，请问是否有更详细的解释？比如您运行的是哪个算法的训练，以及对logits进行奇偶打印的动机。感谢您的补充！

**Comment by jxz7110** [comment]
Align_Anything_Janus/janus/models/modeling_vlm.py 文件中的问题  发现代码写错了  不知道是我理解的问题还是确实是错误的

**Comment by htlou** [comment]
您好，请问您所提供的图1中的logit是在哪个位置print得到的呢？

**Comment by yujinhanml** [comment]
这里似乎有bug，前面的token的位置是better - worse - better padding -worse padding，计算CFG的时候，使用的却是better - better padding 和 worse  -worse padding的加权计算，如果理解错了请纠正

**Comment by htlou** [comment]
> 这里似乎有bug，前面的token的位置是better - worse - better padding -worse padding，计算CFG的时候，使用的却是better - better padding 和 worse -worse padding的加权计算，如果理解错了请纠正

从您的表述看似乎是DPO对的实现逻辑有问题，Janus的DPO分为两种逻辑，但是在处理数据时padding的策略都是一致的（参考`align-anything/datasets/janus/preference.py`中的`PreferenceCollator`，统一是执行(better + better_padding) + (worse + worse_padding)的逻辑，对(better, worse)先padding成(better + better_padding, worse + worse_padding)后再squeeze得到。

如果我的理解有误或者您有后续的问题，欢迎在本issue中进一步提出！
